src/hotspot/share/opto/type.hpp

```diff
@@ -1682,13 +1696,22 @@ inline const TypeAry *Type::isa_ary() const {
   return ((_base == Array) ? (TypeAry*)this : NULL);
 }
 
+inline const TypeVectMask *Type::is_vectmask() const {
+  assert( _base == VectorM, "Not a Vector Mask" );
+  return (TypeVectMask*)this;
+}
+
+inline const TypeVectMask *Type::isa_vectmask() const {
+  return (_base == VectorM) ? (TypeVectMask*)this : NULL;
+}
+
 inline const TypeVect *Type::is_vect() const {
-  assert( _base >= VectorA && _base <= VectorZ, "Not a Vector" );
+  assert( _base >= VectorM && _base <= VectorZ, "Not a Vector" );
   return (TypeVect*)this;
 }
 
 inline const TypeVect *Type::isa_vect() const {
-  return (_base >= VectorA && _base <= VectorZ) ? (TypeVect*)this : NULL;
+  return (_base >= VectorM && _base <= VectorZ) ? (TypeVect*)this : NULL;
```

Comment:
I'm sorry that I still cannot understand why `"TypeVectMask"` is needed to extend from `"TypeVect"`. Is there any other important reasons besides of the logic extension? I'm worried if there is any influence that these two methods treat the `VectorM` as a vector? There are too much usages of these two methods that are specific for vectors especially during register allocation. I'm wondering whether there are any other usages also need the same handles like what you have fixed line 1059 in `x86_32.ad`. Did you have checked all the usages? Thanks!

Reply:
New mask type is indeed a vector of BYTE/BOOLEAN values, thus its ok if it goes though regular flows which processes vector type.

Reply:
OK, I agree to leave it as it for now if we don't have a better idea for TypeVectMask.